scylla/src/frame/request/batch.rs

```diff
@@ -1,29 +1,22 @@
 use anyhow::Result;
 use bytes::{BufMut, Bytes};
+use std::convert::TryInto;
 
 use crate::frame::{
     request::{Request, RequestOpcode},
     types,
-    value::Value,
+    value::BatchValues,
 };
 
-pub struct Batch<'a, I: Iterator<Item = BatchStatementWithValues<'a>> + Clone> {
-    pub statements: I,
-    pub statements_count: usize,
+use crate::statement::batch::BatchStatement;
```

Comment:
There is a problem because `BatchStatement` contains other types of statements so even if moved to `frame` it would still make `frame` depend on `statement`. It makes sense that these modules share the same types because both aim to represent requests somehow. 
Before the change circular dependency was avoided by essentially duplicating `BatchStatement` in `frame` with clever iterators which in my opinion made readability worse than this small circular dependency. But I see that both you and @havaker dislike it so I'm just gonna bring back the iterators in a separate commit which could always be easily reverted.